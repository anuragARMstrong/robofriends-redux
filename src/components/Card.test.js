import { shallow, mount, render } from 'enzyme';
import React from 'react';
import Card from './Card';

// shallow - just renders the component in jsdom env and ignore any nested components
// mount - does the complete rendering of the dom and also invokes react life cycle methods
// render - is more similar to shallow but it has specific usage

// in 90% of the time we use shallow only. because mount can affect other tests as well

it('should render card component', () => {
  // expect(shallow(<Card />).length).toEqual(1);
  expect(shallow(<Card />)).toMatchSnapshot();
});
