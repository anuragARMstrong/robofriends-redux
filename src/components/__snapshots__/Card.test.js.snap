// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card component 1`] = `
<div
  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
>
  <img
    alt="robots"
    height={200}
    src="https://robohash.org/undefined?size=400x400"
    width={200}
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
